server/providers/FantLab.js

```diff
@@ -0,0 +1,172 @@
+const axios = require('axios')
+const Logger = require('../Logger')
+
+class FantLab {
+  // 7 - other
+  // 11 - essay
+  // 12 - article
+  // 22 - disser
+  // 23 - monography
+  // 24 - study
+  // 25 - encyclopedy
+  // 26 - magazine
+  // 46 - sketch
+  // 47 - reportage
+  // 49 - excerpt
+  // 51 - interview
+  // 52 - review
+  // 55 - libretto
+  // 56 - anthology series
+  // 57 - newspaper
+  // types can get here https://api.fantlab.ru/config.json
+  _filterWorkType = [7, 11, 12, 22, 23, 24, 25, 26, 46, 47, 49, 51, 52, 55, 56, 57]
+  _baseUrl = 'https://api.fantlab.ru'
+
+  constructor() { }
+
+  async search(title, author) {
+    let searchString = encodeURIComponent(title)
+    if (author) {
+      searchString += encodeURIComponent(' ' + author)
+    }
+    const url = `${this._baseUrl}/search-works?q=${searchString}&page=1&onlymatches=1`
+    Logger.debug(`[FantLab] Search url: ${url}`)
+    const items = await axios.get(url).then((res) => {
+      return res.data || []
+    }).catch(error => {
+      Logger.error('[FantLab] search error', error)
+      return []
+    })
+
+    return Promise.all(items.map(async item => await this.getWork(item))).then(resArray => {
+      return resArray.filter(res => res != null)
+    })
+  }
+
+  async getWork(item) {
+    const { work_id, work_type_id } = item
+
+    if (this._filterWorkType.includes(work_type_id)) {
+      return { title: null }
```

Comment:
```suggestion
      return null
```
On Line 42 null values are being filtered out.